### PR TITLE
Fix connection loss

### PR DESCRIPTION
**Pull request by antonrohr**
This PR mainly adds support for ssh connection loses. Also some improved error messages, appropriate disabling of actions,  refactoring and general moving around of things. I tried to split it up in multiple commits to make the review easier.

This is tracked by http://b/177305866
This also properly fixes http://b/180327548

### CODE REVIEW

File: src/OrbitQt/orbitmainwindow.cpp
Comment (beckerhe) [inline src/OrbitQt/orbitmainwindow.cpp]: As far as I can see this connection could technically be made twice between the same objects, when SetTarget is called twice. So I would add the Qt::UniqueConnection flag. https://doc.qt.io/qt-5/qt.html#ConnectionType-enum

File: src/OrbitQt/ConnectToStadiaWidget.cpp
Comment (beckerhe) [inline src/OrbitQt/ConnectToStadiaWidget.cpp:8]: Is clangd suggesting that? Looks like an internal header to me. I assume iwyu suggests something else instead.

File: src/OrbitQt/ConnectToStadiaWidget.cpp
Comment (beckerhe) [inline src/OrbitQt/ConnectToStadiaWidget.cpp:93]: From an API design perspective this operation should be reversible, even if the class is currently used only in one situation in the code. The user needs some way to clear the pointer again, even though the one existing user doesn't care. So maybe add a `ClearSshConnectionArtifacts` method which sets the internal pointer to `nullptr`?

File: src/OrbitQt/orbitmainwindow.h
Comment (beckerhe) [inline src/OrbitQt/orbitmainwindow.h:172]: It's a bit unclear if that function is an event handler which is reacting to a connection error or if this is actually an event source which is called from something internal. From the implementation I know it's the former. So maybe something like `OnStadiaConnectionError`?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM.

But it's a large PR. Someone else should approve as well.

File: src/OrbitQt/ProfilingTargetDialog.cpp
Comment (vickyliu-go4it) [inline src/OrbitQt/ProfilingTargetDialog.cpp:505]: May be we can also rename the function `SetInitalStateFromTarget` as `SetStateMachineInitalStateFromTarget` and move this line for setting initial state for `state_machine_` there. If so, we can also have an early return in this method.

**Review by vickyliu-go4it — APPROVED** [review]
Overall look good! I have some minor suggestions, but they are mostly some personal preferences. Please feel free to ignore them. Thanks for making this change! LGTM.

File: src/OrbitQt/orbitmainwindow.cpp
Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.cpp]: I did not know about those. Good idea, I added it.

File: src/OrbitQt/ConnectToStadiaWidget.cpp
Comment (antonrohr) [reply src/OrbitQt/ConnectToStadiaWidget.cpp:92]: done

File: src/OrbitQt/orbitmainwindow.h
Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.h]: done

File: src/OrbitQt/orbitmainwindow.h
Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.h:216]: good idea! I added a comment

File: src/OrbitQt/ProfilingTargetDialog.cpp
Comment (antonrohr) [reply src/OrbitQt/ProfilingTargetDialog.cpp]: I agree it might be confusing. I added the `StateMachine` part of the name

File: src/OrbitQt/ProfilingTargetDialog.cpp
Comment (antonrohr) [reply src/OrbitQt/ProfilingTargetDialog.cpp]: I changed it. ptal

Comment (vickyliu-go4it) [reply src/OrbitQt/ProfilingTargetDialog.cpp]: Looks good. Thanks for making the change : )

File: src/OrbitQt/ConnectToStadiaWidget.cpp
Comment (antonrohr) [reply src/OrbitQt/ConnectToStadiaWidget.cpp]: Yes it was clangd. iwyu did not complain when it was in there. I removed it and everything still works, also iwyu is not complaining as far as I can tell

**Comment by antonrohr** [comment]
@beckerhe I would rebase and merge this soon, if there are no objections.

**Comment by beckerhe** [comment]
> @beckerhe I would rebase and merge this soon, if there are no objections.

Sry, yes. Sure.

**Pull request merged**
